kadi/events/models.py

```diff
@@ -1114,6 +1117,121 @@ class Eclipse(TlmEvent):
     fetch_event_msids = ['aoeclips', 'eb1k5', 'eb2k5', 'eb3k5']
 
 
+class StarAcq(TlmEvent):
+    """
+    ACA star acquisition
+
+    **Event definition**: interval where ``AOACASEQ == 'AQXN'``
+
+    **Fields**
+
+    ======== ========== ================================
+     Field      Type              Description
+    ======== ========== ================================
+      start   Char(21)   Start time (YYYY:DDD:HH:MM:SS)
+       stop   Char(21)    Stop time (YYYY:DDD:HH:MM:SS)
+     tstart      Float            Start time (CXC secs)
+      tstop      Float             Stop time (CXC secs)
+        dur      Float                  Duration (secs)
+    ======== ========== ================================
+    """
+    event_msids = ['aoacaseq']
+    event_val = 'AQXN'
+    event_min_dur = 20
+    lookback = 3
+
+    n_id = models.IntegerField(help_text='Number of identified acquisition stars', null=True)
+    id_status = models.CharField(max_length=8, null=True,
+                                 help_text='Identification status of catalog entries (T or F)')
+    one_shot_roll = models.FloatField(help_text='One shot update in roll (arcsec)', null=True)
+    one_shot_pitch = models.FloatField(help_text='One shot update in pitch (arcsec)', null=True)
+    one_shot_yaw = models.FloatField(help_text='One shot update in yaw (arcsec)', null=True)
+
+    @classmethod
+    def get_alerts(cls, event, event_msidset):
+        """
+        Check for large one-shot in pitch or yaw, or a small number of identified acq
+        stars.
+        """
+        def get_obsid():
+            dat = fetch.Msid('cobsrqid', event['tstop'] - 10, event['tstop'] + 300)
+            return int(dat.vals[0])
+
+        alerts = []
+
+        if abs(event['one_shot_pitch']) > 60 or abs(event['one_shot_yaw']) > 60:
+            obsid = get_obsid()
+            message = ('One shot at {}\n'
+                       'Roll  : {:.2f}\n'
+                       'Pitch : {:.2f}\n'
+                       'Yaw   : {:.2f}\n'
+                       '\n'
+                       'See http://kadi.cfa.harvard.edu/mica/?obsid_or_date={}\n'
+                       .format(event['stop'],
+                               event['one_shot_roll'],
+                               event['one_shot_pitch'],
+                               event['one_shot_yaw'],
+                               obsid))
+            alerts.append({'subject': 'Large one-shot update at {} (obsid {})'
+                           .format(event['stop'], obsid),
+                           'message': message})
+
+        if event['n_id'] <= 5:
```

Comment:
I'd be comfortable setting this to <= 4 at this point (assuming we keep up with performance trending).